lib/src/block_parser.dart

```diff
@@ -571,6 +574,86 @@ class OrderedListSyntax extends ListSyntax {
   const OrderedListSyntax();
 }
 
+/// Parses tables.
+class TableSyntax extends BlockSyntax {
+  bool get canEndBlock => false;
+
+  const TableSyntax();
+
+  bool canParse(BlockParser parser) {
+    // Note: matches *next* line, not the current one. We're looking for the
+    // bar separating the head row from the body rows.
+    return parser.matchesNext(_tablePattern);
+  }
+
+  /// Parses a table into its three parts:
+  ///
+  /// * a head row of head cells (`<th>` cells)
+  /// * a divider of hyphens and pipes (not rendered)
+  /// * many body rows of body cells (`<td>` cells)
+  Node parse(BlockParser parser) {
+    var alignments = parseAlignments(parser.next);
+    var head = new Element('thead', [parseRow(parser, alignments, 'th')]);
+
+    // Advance past the divider of hyphens.
+    parser.advance();
+
+    var rows = <Element>[];
+    while (!parser.isDone && !parser.matches(_emptyPattern)) {
+      rows.add(parseRow(parser, alignments, 'td'));
+    }
+    var body = new Element('tbody', rows);
+
+    return new Element('table', [head, body]);
+  }
+
+  List<String> parseAlignments(String line) {
+    line = line
+        .replaceFirst(new RegExp(r'^\|'), '')
+        .replaceFirst(new RegExp(r'\|$'), '');
+    return line.split('|').map((column) {
+      if (column.startsWith(':') && column.endsWith(':')) return 'center';
+      if (column.startsWith(':')) return 'left';
+      if (column.endsWith(':')) return 'right';
+      return null;
+    }).toList();
+  }
+
+  Node parseRow(BlockParser parser, List<String> alignments, String cellType) {
+    var line = parser.current
+        .replaceFirst(new RegExp(r'^\|\s*'), '')
+        .replaceFirst(new RegExp(r'\s*\|$'), '');
+    var contents = parser.document.parseInline(line);
```

Comment:
This one is bizarre. According to Babelmark, GitHub, Pandoc, PHP Markdown Extra, kramdown (3rd most popular Rubygem):

* When considering [strong emphasis](http://johnmacfarlane.net/babelmark2/?normalize=1&text=%7C+Header+%7C+Header+%7C%0A%7C+------+%7C+------+%7C%0A%7C+**Cell+%7C+Cell**+%7C), emphasis, links: They split on cells first, then parse inline.
* When considering [backticks](http://johnmacfarlane.net/babelmark2/?normalize=1&text=%7C+Header+%7C+Header+%7C%0A%7C+------+%7C+------+%7C%0A%7C+%60Cell+%7C+Cell%60+%7C): They parse inline first, then split cells.

You can see here:

    | Header |   Header    | Header |
    | ------ | ----------- | ------ |
    | `Cell  | Cell` *Cell | Cell*  |

is parsed by GitHub as:

| Header | Header | Header |
| ------ | ------ | ------ |
| `Cell | Cell` *Cell | Cell* |

In light of this, I've split first, then parse inlines, and match most of the cases.